[AOTI] Change the cpp wrapper codegen for sdpa

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #120714
* #120513
* #120629
* __->__ #120592
* #119987

Summary: Switch codegen for sdpa to always point to v2 in the C shim. Since aoti_torch__scaled_dot_product_flash_attention_v2 has been introduced for a while, there shouldn't be any FC issue in production.

Differential Revision: [D54258090](https://our.internmc.facebook.com/intern/diff/D54258090)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames

desertfire committed:
[AOTI] Change the cpp wrapper codegen for sdpa

Summary: Switch codegen for sdpa to always point to v2 in the C shim. Since aoti_torch__scaled_dot_product_flash_attention_v2 has been introduced for a while, there shouldn't be any FC issue in production.

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54258090).

desertfire committed:
Update on "[AOTI] Change the cpp wrapper codegen for sdpa"


Summary: Switch codegen for sdpa to always point to v2 in the C shim. Since aoti_torch__scaled_dot_product_flash_attention_v2 has been introduced for a while, there shouldn't be any FC issue in production.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler amjames

Differential Revision: [D54258090](https://our.internmc.facebook.com/intern/diff/D54258090)

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54258090).

desertfire committed:
Update on "[AOTI] Change the cpp wrapper codegen for sdpa"


Summary: Switch codegen for sdpa to always point to v2 in the C shim. Since aoti_torch__scaled_dot_product_flash_attention_v2 has been introduced for a while, there shouldn't be any FC issue in production.

Differential Revision: [D54258090](https://our.internmc.facebook.com/intern/diff/D54258090)

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54258090).

chenyang78 commented on torch/_inductor/ir.py:
```diff
@@ -4772,15 +4772,8 @@ def _get_abi_compatible_kernel(self):
+            # For sdpa, we need the v2 version since v1 didn't consider optional arg
+            return f"{self.cpp_kernel_name}_v2"
 
         kernel_to_ver = {"at::_scaled_dot_product_flash_attention": sdpa_ver_fn}
```
Since we only have a single version now, perhaps we could just get rid of the entire ```sdpa_ver_fn``` logic.

desertfire committed:
Update on "[AOTI] Change the cpp wrapper codegen for sdpa"


Summary: Switch codegen for sdpa to always point to v2 in the C shim. Since aoti_torch__scaled_dot_product_flash_attention_v2 has been introduced for a while, there shouldn't be any FC issue in production.

Differential Revision: [D54258090](https://our.internmc.facebook.com/intern/diff/D54258090)

[ghstack-poisoned]

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54258090).

chenyang78 reviewed (approved):
LGTM. Thanks!

desertfire commented:
@desertfire has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D54258090).
